src/Microsoft.ML.DataView/KeyDataViewType.cs

```diff
@@ -3,55 +3,73 @@
 // See the LICENSE file in the project root for more information.
 
 using System;
-using Microsoft.ML.Internal.Utilities;
-using Microsoft.ML.Runtime;
 
 namespace Microsoft.ML.Data
 {
     /// <summary>
-    /// KeyTypes are for "id"-like data. The information happens to be stored in an unsigned integer
+    /// KeyDataViewTypes are for "id"-like data. The information happens to be stored in an unsigned integer
     /// type, but the information is not inherently numeric, so, typically, arithmetic is not
     /// meaningful. Examples are SSNs, phone numbers, auto-generated/incremented key values,
     /// class numbers, etc. For example, in multi-class classification, the label is typically
-    /// a class number which is naturally a KeyType.
+    /// a class number which is naturally a KeyDataViewType.
     ///
-    /// KeyTypes have a cardinality (i.e., <see cref="Count"/>) that is strictly positive.
+    /// KeyDataViewTypes have a cardinality (i.e., <see cref="Count"/>) that is strictly positive.
     ///
     /// Note that the underlying representation value does not necessarily match the logical value.
-    /// For example, if a KeyType has range 0-5000, then it has a <see cref="Count"/> of 5001, but
+    /// For example, if a KeyDataViewType has range 0-5000, then it has a <see cref="Count"/> of 5001, but
     /// the representational values are 1-5001. The representation value zero is reserved
     /// to mean a missing value (similar to NaN).
     /// </summary>
-    public sealed class KeyType : PrimitiveDataViewType
+    public sealed class KeyDataViewType : PrimitiveDataViewType
     {
-        public KeyType(Type type, ulong count)
+        public KeyDataViewType(Type type, ulong count)
```

Comment:
>KeyDataViewType [](start = 15, length = 15)

Hi @eric.erhardt  - There are a couple of things that I am not crazy about the name KeyDataViewType. First that it has the word Type in the name (I know, I know KeyType does too and I am not saying that name is better), but there are a lot of classes that do not have the name Type in it  (File, Socket, etc) where the assumption that the object is a Type is implied.

Also having DataView in the name associates that it is used with DataViews -- but this is where namespaces semi-manages that grouping and doesn't need to be added to the class name. I think it would be great to name it more along the lines of what it does -- maybe Identifier or KeyIdentifier? 

I could be missing a whole lot of context here -- like maybe there is thread where this name was agreed upon or maybe this came out of an api review. So if that's the case disregard all of this because that means I could have added feedback prior to the PR :)

Reply:
> First that it has the word Type in the name 

These things _are_ types. They represent what type a column in an IDataView is. They are metadata that is part of the schema class.

> having DataView in the name associates that it is used with DataViews

The thing we are differentiating from is explicitly `System.Type` - which is a .NET Type concept. DataViewTypes and .NET System.Types have a relationship, but they are not 1:1. Since `KeyType` and `VectorType` inherit from the abstract base `DataViewType` class, we decided in #2297 to put the `DataViewType` suffix on all these types. I just happened to miss these two.  (And come to think of it - I missed 

https://github.com/dotnet/machinelearning/blob/70ef7ecd43b031b481a4047ea361da5e2f360336/src/Microsoft.ML.ImageAnalytics/ImageType.cs#L12

as well.  I logged #3050 for that.